src/cms/views/media/media_actions.py

```diff
@@ -1,38 +1,274 @@
+        Q(region=region) | Q(region__isnull=True), parent=directory
+    )
+
+    result = list(map(lambda directory: directory.serialize(), directories)) + list(
+        map(lambda document: document.serialize(), documents)
+    )
+
+    return JsonResponse({"success": True, "data": result}, status=200)
+
+
+@login_required
+@region_permission_required
+@require_POST
+def edit_media_element_ajax(request, region_slug=None):
+    """
+    View provides the edit of a file via AJAX.
+
+    :param request: The current request
+    :type request: ~django.http.HttpResponse
+
+    :param region_slug: The slug of the current region
+    :type region_slug: str
+
+    :raises ~django.core.exceptions.PermissionDenied: If the user does not have permissions for this region.
+
+    :return: On success status 200
+    :rtype: ~django.http.JsonResponse
+    """
+    region = None
+    if region_slug is not None:
+        region = Region.objects.get(slug=region_slug)
+
+    # Check for correct Request. As we are updating a database object, only POST is allowed.
+    if request.method != "POST":
+        return JsonResponse({"error": "Invalid request."}, status=405)
+
+    json_data = json.loads(request.body.decode("utf-8"))
+    media_element_id = json_data["id"]
+    media_element = get_object_or_404(Document, id=media_element_id)
+    if media_element.region != region:
+        raise PermissionDenied
+
+    media_element.name = json_data["name"]
+    media_element.alt_text = json_data["alt_text"]
+    media_element.save()
+
+    return JsonResponse({"success": True})
+
+
+@login_required
+@region_permission_required
+@require_POST
+def create_directory_ajax(request, region_slug=None):
+    """
+    View provides the frontend with the option to create a directory via AJAX.
+
+    :param request: The current request
+    :type request: ~django.http.HttpResponse
+
+    :param region_slug: The slug of the current region
+    :type region_slug: str
+
+    :raises ~django.core.exceptions.PermissionDenied: If the user does not have permissions for this region.
+
+    :return: On success returns the directory content and status 200
+    :rtype: ~django.http.JsonResponse
+    """
+    region = None
+    if region_slug is not None:
+        region = Region.objects.get(slug=region_slug)
+    # Check for correct Request. As we are updating a database object, only POST is allowed.
+    if request.method != "POST":
+        return JsonResponse({"error": "Invalid request."}, status=405)
+
+    json_data = json.loads(request.body.decode("utf-8"))
+    parent_directory = None
+    if int(json_data["parentDirectory"]) != 0:
+        parent_directory = get_object_or_404(Directory, id=json_data["parentDirectory"])
+
+        if parent_directory.region != region:
+            raise PermissionDenied
+    else:
+        json_data["parentDirectory"] = None
+
+    if Directory.objects.filter(
+        parent=json_data["parentDirectory"], name=json_data["directoryName"]
```

Comment:
```suggestion
    queryset = Directory.objects
    if region:
        # If directory is created for a region, only limit choices to the global library and the regional one
        queryset = queryset.filter(Q(region=region) | Q(region__isnull=True))
    if queryset.filter(
        parent=parent_directory,
        name=json_data["directoryName"],
```
I think it's enough if the directory name is unique across the global library and the region's library...
If two separate regions want to create directories with the same name, that should be fine I think...